File: Segundo/Monopoly/civitas_java (1)/proyecto_civitas/src/civitas/CasillaSorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

/**
 *
 * @author pc
 */
import java.util.ArrayList;
public class CasillaSorpresa extends Casilla {
    private Sorpresa sorpresa;
    private MazoSorpresas mazo;
    
    CasillaSorpresa(MazoSorpresas mazo,String nombre){
        super(nombre);
        this.mazo=mazo;
    }
    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos)){
            sorpresa=mazo.siguiente();
            informe(actual,todos);
            sorpresa.aplicarAJugador(actual, todos);
            
        }
    }
    @Override
    public String toString() {
        return "Casilla Sorpresa: "+ sorpresa + "\nMazo: " + mazo ;
    }
}

File: Segundo/Monopoly/civitas_ruby (1)/proyecto_civitas_ruby/lib/casilla_sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
class CasillaSorpresa < Casilla
  def initialize(nombre,mazo)
    super(nombre)
    @mazo = mazo
  end
  def recibeJugador(actual,todos)
    if(jugadorCorrecto(actual, todos))
      super(actual,todos)
      sorpresa1=@mazo.Siguiente
      sorpresa1.aplicarAJugador(actual, todos)
    end
  end
   
  def toString
    cadena="Casilla: #{@nombre} \n\n"
    return cadena
  end
   public :toString
end
end